sdks/python/src/opik/evaluation/metrics/llm_judges/structure_output_compliance/metric.py

```diff
@@ -0,0 +1,95 @@
+from typing import Union, Optional, List, Any
+import pydantic
+
+from opik.evaluation.models import base_model, models_factory
+from opik.evaluation.metrics import score_result, base_metric
+from . import template, parser
+
+
+class StructuredOutputComplianceResponseFormat(pydantic.BaseModel):
+    score: bool
+    reason: List[str]
+
+
+class StructuredOutputCompliance(base_metric.BaseMetric):
+    """
+    Metric to evaluate whether an LLM's output complies with a specified structured format.
+    This includes checking for valid JSON, JSON-LD compatibility, or adherence to a provided
+    Pydantic/JSON schema.
+    Args:
+        model: LLM to use for evaluation. Can be a string or an OpikBaseModel instance.
+        name: Metric name.
+        few_shot_examples: Optional few-shot examples to guide the LLM's judgment.
+        track: Whether to track metric execution for observability.
+        project_name: Optional name for tracking in an observability tool.
+    """
+
+    def __init__(
+        self,
+        model: Optional[Union[str, base_model.OpikBaseModel]] = None,
+        name: str = "structured_output_compliance",
+        few_shot_examples: Optional[List[template.FewShotExampleStructuredOutputCompliance]] = None,
+        track: bool = True,
+        project_name: Optional[str] = None,
+    ):
+        super().__init__(name=name, track=track, project_name=project_name)
+        self._init_model(model)
+        self.few_shot_examples = few_shot_examples
+
+    def _init_model(self, model: Optional[Union[str, base_model.OpikBaseModel]]) -> None:
+        if isinstance(model, base_model.OpikBaseModel):
+            self._model = model
+        else:
+            self._model = models_factory.get(model_name=model)
+
+    def score(
+        self,
+        output: str,
+        schema: Optional[str] = None,
+        **ignored_kwargs: Any,
+    ) -> score_result.ScoreResult:
+        """
+        Synchronously compute the structured output compliance score.
+        Args:
+            output: The LLM's output to validate.
+            schema: Optional JSON or Pydantic schema to validate against.
+        Returns:
+            score_result.ScoreResult: An object containing the compliance score and reasons.
+        """
+        llm_query = template.generate_query(
+            output=output,
+            schema=schema,
+            few_shot_examples=self.few_shot_examples,
+        )
+
+        model_output = self._model.generate_string(
+            input=llm_query, response_format=StructuredOutputComplianceResponseFormat
+        )
+
+        return parser.parse_model_output(content=model_output, name=self.name)
+
+    async def ascore(
```

Comment:
This should be covered by unit tests with mocked model and at least one integration test with real model.